fixes #6227

rpbouman opened the pull request:
- adds a test that demonstrates old implementation of getTables always throws an SQLException
- rewrites getTables to provide handling for the catalog parameter.

rpbouman committed:
fixes #6227
- adds a test that demonstrates old implementation of getTables always throws an SQLException
- rewrites getTables to provide handling for the catalog parameter.

Mytherin reviewed (approved):
Thanks for the PR! LGTM. Perhaps @Mause can also have a look?

Mytherin commented on tools/jdbc/src/test/java/org/duckdb/test/TestDuckDBJDBC.java:
```diff
@@ -1821,6 +1821,18 @@ public static void test_schema_reflection() throws Exception {
+	public static void test_get_tables_with_catalog() throws Exception {
+		Connection conn = DriverManager.getConnection("jdbc:duckdb:");
+		String catalog = conn.getCatalog();
+		DatabaseMetaData databaseMetaData = conn.getMetaData();
```
Could we add some tests that verify the `getTables` correctly takes the catalog into account, by e.g. attaching an in-memory database and creating tables there?

rpbouman replied:
Sure! Will pick it up tonight.

rpbouman replied:
@Mytherin - something like this? https://github.com/duckdb/duckdb/pull/6230/commits/df0303c652e2b874ec5005b7faf712e80f47dee4

rpbouman committed:
- renamed test_get_tables_with_catalog to test_get_tables_with_current_catalog, explicitly test catalog in resultset
- added test_get_tables_with_attached_catalog to test with attached catalog, as well as with null argument (all catalogs)

Mytherin commented:
Looks good - thanks!
